### PR TITLE
improve the helm chart deployment method changed logic

### PR DESCRIPTION
**Pull request by laverya**
now it will only error for a change if the chart was not previously excluded

#### What this PR does / why we need it:
This PR makes it so that an 'excluded' helm chart can have its deployment method changed without an error.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
* excluded helm charts can change between 'UseHelmInstall: true' and 'UseHelmInstall: false' without errors, bypassing the normal check.
```

### CODE REVIEW

File: pkg/pull/pull.go [hunk 1]
```diff
@@ -292,6 +292,10 @@ func Pull(upstreamURI string, pullOptions PullOptions) (string, error) {
 		}
 
 		for _, prevChart := range prevHelmCharts {
+			if prevChart.Spec.Exclude.BoolVal || prevChart.Spec.Exclude.String() == "1" {
```
Comment (MikaelSmith) [inline pkg/pull/pull.go:295]: Feels like there should be Truthy/Falsey methods to avoid having to copy this pattern around.

Comment (laverya) [reply pkg/pull/pull.go:295]: yeah this isn't how I should actually implement it, in retrospect

I should just be using the .String method, for one

Comment (laverya) [reply pkg/pull/pull.go]: because .String already does the conversion of true/false into "1"/"0"

**Review by morningvera — DISMISSED** [review]

**Comment by MikaelSmith** [comment]
Anything blocking this?

**Review by divolgin — APPROVED** [review]

File: pkg/pull/pull.go [hunk 2]
```diff
@@ -292,6 +292,10 @@ func Pull(upstreamURI string, pullOptions PullOptions) (string, error) {
 		}
 
 		for _, prevChart := range prevHelmCharts {
+			if prevChart.Spec.Exclude.String() == "1" {
```
Comment (MikaelSmith) [inline pkg/pull/pull.go:295]: What is prevChart here? The currently deployed version or the last rendered version?

Comment (laverya) [reply pkg/pull/pull.go]: I believe it's the last rendered version, but I am not sure right now - @divolgin ?

**Review by divolgin — APPROVED** [review]

File: pkg/pull/pull.go (hunk 2 above)
Comment (divolgin) [reply pkg/pull/pull.go]: it's the previous rendered version, `sequence - 1`

Comment (MikaelSmith) [reply pkg/pull/pull.go]: Seems like it'd be better to compare to the actually deployed version, but I don't know how hard that is to do.

**Pull request merged**